### DISCUSSION TITLE
Hub content type available in site or list

### DISCUSSION THREAD

**Discussion by matjazreberc** [pnp/pnpcore#1557] [Q&A]
Hello,
I am adding Content type from hub to site  / list,
and sometimes happens that the process will not finish, it try to add but don't stop.

I only see this output, no Exception is thrown.

Start processing HTTP request GET https://graph.microsoft.com/v1.0/sites/***/operations/contentTypeCopy,0x010***
[2024-10-17T05:10:18.049Z] Sending HTTP request GET https://graph.microsoft.com/v1.0/sites/***/operations/contentTypeCopy,0x010***
[2024-10-17T05:10:18.171Z] Received HTTP response headers after 189.7049ms - 200
[2024-10-17T05:10:18.172Z] End processing HTTP request after 190.7999ms - 200

I'm using code from [PNP Core SDK](https://pnp.github.io/pnpcore/using-the-sdk/contenttypes-hub.html#making-a-content-type-from-the-hub-available-in-a-site-or-list)  **Making a content type from the hub available in a site or list**

Before adding content type, I'm creating site collection, if it not exist and then creating document library.
What could be the reason for this?
Thank you, Matjaz

**Comment by jansenbe** [comment]
@matjazreberc : does the issue also happen when the site collection was created a while ago? Site collection creation nowadays is fast (compared to earlier days when it could take 10 minutes)...but one of the reasons for it being fast is because we do some steps async. If you're code creates the site collection and immediately after tries to pull the CT from the hub it might simply be too fast. This is a guess from my side, but maybe it can help
